src/components/PostDetails.tsx

```diff
@@ -1,106 +1,121 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment, CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+
+type Props = {
+  isCommentLoading: boolean;
+  postComments: Comment[];
+  selectedPost: Post | null;
+  errorsComment: boolean;
+  onDeleteComment: (commentId: number) => void;
+  onAddComment: (newComment: CommentData) => void;
+  isButtonAddLoading: boolean;
+  onShowCommentFormChange: (item: boolean) => void;
+  isCommentFormVisible: boolean;
+  userSelected: User | null;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  isCommentLoading,
+  postComments,
+  selectedPost,
+  errorsComment,
+  onDeleteComment,
+  onAddComment,
+  isButtonAddLoading,
+  onShowCommentFormChange,
+  isCommentFormVisible,
+  userSelected,
+}) => {
+  useEffect(() => {
+    onShowCommentFormChange(false);
```

Comment:
You are initializing your state with props. This is not recommended as it can lead to bugs as your state will only be initialized with the props at the time of mounting. This means if the props change, your state will not update. You should use the props directly or use a method to update the state when the props change.